pkg/partmgr/partmgr.go

```
@@ -18,123 +18,203 @@ under the Apache 2.0 license is conditioned upon your compliance with
 such restriction.
 */
 
-// TODO: need to expand from a single partition to multiple
-
 package partmgr
 
 import (
+	"encoding/json"
 	"fmt"
+	"github.com/pkg/errors"
+	"github.com/v3io/v3io-go-http"
 	"github.com/v3io/v3io-tsdb/pkg/aggregate"
 	"github.com/v3io/v3io-tsdb/pkg/config"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"math"
+	"path"
+	"strconv"
 	"sync"
 )
 
-// Create new Partition Manager, for now confined to one Cyclic partition
-func NewPartitionMngr(cfg *config.DBPartConfig, path string) *PartitionManager {
-	newMngr := &PartitionManager{cfg: cfg, path: path, cyclic: true, ignoreWrap: true}
-	newMngr.headPartition = NewDBPartition(newMngr)
+// Create new Partition Manager
+func NewPartitionMngr(cfg *config.Schema, partPath string, cont *v3io.Container) *PartitionManager {
+	currentPartitionInterval, _ := utils.Str2duration(cfg.PartitionSchemaInfo.PartitionerInterval)
```

do not ignore errors